### PR TITLE
MINOR: Add junit properties to display parameterized test names

### PR DESCRIPTION
**Pull request by alok123t**
In many parameterized tests, the display name is broken. Example - `testMetadataFetch` appears as `[1] true`, `[2] false`  [link](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-14607/9/testReport/junit/org.apache.kafka.clients.producer/KafkaProducerTest/) 
This is because the constant in `@ParameterizedTest`
```java
String DEFAULT_DISPLAY_NAME = "[{index}] {argumentsWithNames}";
```

This PR adds a new `junit-platform.properties` which overrides to add a `{displayName}` which shows the `the display name of the method`

For existing tests which override the name, should work as is. The precedence rules are explained

> 1. `name` attribute in `@ParameterizedTest`, if present
> 2. value of the `junit.jupiter.params.displayname.default` configuration parameter, if present
> 3. `DEFAULT_DISPLAY_NAME` constant defined in `@ParameterizedTest`

Source: https://junit.org/junit5/docs/current/user-guide/#writing-tests-parameterized-tests-display-names

Sample test run output 
Before: `[1] true` [link](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-14607/9/testReport/junit/org.apache.kafka.clients.producer/KafkaProducerTest/)
After: `testMetadataExpiry(boolean).false` [link](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-14687/1/testReport/junit/org.apache.kafka.clients.producer/KafkaProducerTest/)

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by divijvaidya** [comment]
This is a great change but note that we don't use JUnit5 for all tests. Hence we need to ensure that this change doesn't impact Junit4 tests. Ref:
https://github.com/apache/kafka/blob/9911fab1a1a95b7783c21fef75f043ea42fea790/build.gradle#L470 


Also, can we do it for all tests in Kafka instead of just for the clients module?

**Comment by alok123t** [comment]
@divijvaidya added in modules, where `ParameterizedTest` is currently used as part of https://github.com/apache/kafka/pull/14687/commits/777d14d3aba7c59070916497388d82adeb19405a. We could also just do it in all modules as well, let me know if that's useful

**Comment by alok123t** [comment]
Thanks for the context on JUnit4. Looks like streams/connect today use this, example [ConsistencyVectorIntegrationTest](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-14687/1/testReport/junit/org.apache.kafka.streams.integration/ConsistencyVectorIntegrationTest/). 
I'll check the CI output if we see any discrepancies before/after

**Comment by divijvaidya** [comment]
@alok123t Let me know when this is ready for review. And yes, let's just add this proactively to all modules OR, is it possible to put it at one place and apply for all modules (in build.gradle you will notice things such as rat are configured once and get applied for all modules)? Perhaps something worth considering.

**Comment by alok123t** [comment]
@divijvaidya looks like we only need add the junit file in root modules like `server-common` and `clients` and other should be able to import

**Comment by divijvaidya** [comment]
> @divijvaidya looks like we only need add the junit file in root modules like `server-common` and `clients` and other should be able to import

Sweet! Please verify after the CI is complete that we don't have regression with tests which use Junit 4 and then we should be ready to merge.

**Comment by alok123t** [comment]
@divijvaidya I think it looks reasonable JUnit 4 test - [ConsistencyVectorIntegrationTest](
https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-14687/5/testReport/junit/org.apache.kafka.streams.integration/ConsistencyVectorIntegrationTest/) running on CI

**Comment by divijvaidya** [comment]
I triggered a re-build of the CI since JDK11+Scala 2.13 was failing

**Comment by divijvaidya** [comment]
We have an unusually large number of failures. @alok123t can you please check the failures to see if they might be related to this change?

**Comment by alok123t** [comment]
@divijvaidya looks unrelated. Most failed tests are related to `RestoreIntegrationTest` - likely some regression when I merged in latest changes from trunk to trigger a new build

**Comment by dajac** [comment]
Hey @alok123t. Thanks for working on this. In the last build, I just noticed that parameterized still are still reported as:
```
Build / JDK 21 and Scala 2.13 / [5] Type=ZK, MetadataVersion=3.7-IV1, Security=PLAINTEXT – kafka.zk.ZkMigrationIntegrationTest
```
Shouldn't your patch fix those as well?

**Comment by dajac** [comment]
@alok123t Any progress on this one? I actually need it to continue my [experiment](https://github.com/apache/kafka/pull/14743) to improve our CI.

**Comment by alok123t** [comment]
> Hey @alok123t. Thanks for working on this. In the last build, I just noticed that parameterized still are still reported as:
> 
> ```
> Build / JDK 21 and Scala 2.13 / [5] Type=ZK, MetadataVersion=3.7-IV1, Security=PLAINTEXT – kafka.zk.ZkMigrationIntegrationTest
> ```
> 
> Shouldn't your patch fix those as well?

@dajac Looks like these tests are based on the `@ClusterTest` annotation which internally overrides the `name()` https://github.com/apache/kafka/blob/af747fbfed7e81617c3b3ad0e4dc8c857aa9502b/core/src/test/java/kafka/test/annotation/ClusterTest.java#L41 and has it own custom implementation of overriding the name https://github.com/apache/kafka/blob/9b468fb278701be836a2641650356907bf84860a/core/src/test/java/kafka/test/junit/ClusterTestExtensions.java#L187
In case there are no other outstanding comments, shall we go ahead and merge this PR and we can fix the naming for the ones which use annotations in a subsequent PR

**Comment by dajac** [comment]
Thanks for the explanation, @alok123t! I think that we could fix those separately, if we want to. @divijvaidya Do you want to do a final review?

**Review by dajac — APPROVED** [review]
LGTM, thanks!

**Comment by dajac** [comment]
We need to investigate the failed tests before we can merge this one.

File: streams/src/test/java/org/apache/kafka/streams/integration/utils/IntegrationTestUtils.java
Comment (dajac) [inline streams/src/test/java/org/apache/kafka/streams/integration/utils/IntegrationTestUtils.java]: With the current method, we end up with a non-unique name for the parameterized tests. Therefore, I suggest to just use the name of the method with a uuid.

File: streams/src/test/java/org/apache/kafka/streams/integration/HighAvailabilityTaskAssignorIntegrationTest.java
Comment (dajac) [inline streams/src/test/java/org/apache/kafka/streams/integration/HighAvailabilityTaskAssignorIntegrationTest.java]: The name got too long (above the max for a topic name) with the changes in this PR. As the `appId` already contains the `testId`, it seems that we could just remove it.

**Review by cadonna — APPROVED** [review]
LGTM!

File: streams/src/test/java/org/apache/kafka/streams/integration/HighAvailabilityTaskAssignorIntegrationTest.java
Comment (cadonna) [inline streams/src/test/java/org/apache/kafka/streams/integration/HighAvailabilityTaskAssignorIntegrationTest.java]: Why does `testId` not work here?

Comment (dajac) [reply streams/src/test/java/org/apache/kafka/streams/integration/HighAvailabilityTaskAssignorIntegrationTest.java]: This change is actually not needed anymore with the last version of `safeUniqueTestName`. Let me remove it. But to answer your question, the topic name derived from the store name was too long because the testId was included twice.

**Comment by dajac** [comment]
I cannot get a clean build for this one...

File: streams/src/test/java/org/apache/kafka/streams/integration/utils/IntegrationTestUtils.java
Comment (cadonna) [inline streams/src/test/java/org/apache/kafka/streams/integration/utils/IntegrationTestUtils.java]: Why not just replacing `-` with a different character?

Comment (dajac) [reply streams/src/test/java/org/apache/kafka/streams/integration/utils/IntegrationTestUtils.java]: Yeah, I considered this as well. The question is which one should we use? We could use `_` but then the uuid looks a bit weird when you get one with `_` in the middle. I thought that it is ok to re-generate a new one in this case. I can change it if you prefer.

Comment (cadonna) [reply streams/src/test/java/org/apache/kafka/streams/integration/utils/IntegrationTestUtils.java]: I think we can use whatever we want, also `_`. There is an override of `safeUniqueTestName()` that does a similar replacement. I think you could add there `-`. It just seemed costly and somehow overly complicated to me to re-generate a random sequence just to get rid of a character. But probably it is not costly because in most cases we only need to re-generate once.

Comment (dajac) [reply streams/src/test/java/org/apache/kafka/streams/integration/utils/IntegrationTestUtils.java]: Addressed with the last commit.

**Review by cadonna — APPROVED** [review]

**Pull request merged**
